### DISCUSSION TITLE
tutorials/lift-off-part1/09-defining-a-query

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#40] [course feedback]
# tutorials/lift-off-part1/09-defining-a-query

https://www.apollographql.com/tutorials/lift-off-part1/09-defining-a-query

**Comment by CMH7** [comment]
Got no errors so far!

**Comment by anvodev** [comment]
Excellent experience so far!

**Comment by zibiv** [comment]
Why in the video we use gql from @apollo/client, but in the text from codegen
> We can now paste the query directly into our empty gql function.
And put the query in the gql string, not in the function...

**Reply by mabuyo** [reply]
Hey there @zibiv - please note that your language selection will be applied to code blocks and exercises. Videos may not yet reflect your preference. Apologies for the confusion!

**Reply by zibiv** [reply]
You right, my language is TS and it will be applied to code blocks and exercises. But in this part of the text: "We can now paste the query directly into our empty gql function." we get code section with tagged template gql`query` (gql form "@apollo/client", not gql function from our codegen) =)

**Reply by mabuyo** [reply]
Ahh yes, I see that now! I will fix that shortly, thanks for letting us know!

**Comment by zibiv** [comment]
And if you get error "Unable to find any GraphQL type definitions for the following pointers:" try to add . to document property of config  documents: ["./src/**/*.tsx"]

**Reply by mabuyo** [reply]
Yes, thank you! This was covered in the [previous lesson for codegen](https://www.apollographql.com/tutorials/lift-off-part1/09-codegen), but might be helpful for us to add a troubleshooting section here for those that missed it.

**Comment by cdxx-lxix** [comment]
Hello there!
There is a typo:
```
import { __gql__ } from "@apollo/client";
const TRACKS = __gq__`
  # Query goes here
`;
```
The "l" is gone but in the next code block it's again there.

**Reply by lizhennessy** [reply]
Thanks so much for bringing this to our attention, @cdxx-lxix ! We will get this fixed right away.

**Comment by SteveBold** [comment]
I managed to add the required detail for the challenge into the schema:
type Query {
    "Get Space Cat Missions"
    spaceCats: [SpaceCat!]!
  }

"A track is a group of Modules that teaches about a specific topic"
type SpaceCat {
    "name of the cat"
    name: String!
    "age of the cat"
    age: Int!
    " mission information for the cat"
    missions: [Mission]!
}  

"Author of a complete Track or a Module"
type Mission {
    "name of the mission"
    name: String!
    "description of the mission"
    description: String!
}

Also the changes to the index.js as follows:

  Query: () => ({
      spaceCats: () => [
        // Mock data for the first space cat
        {
          name: 'Astro Kitty',
          age: 3,
          missions: [
            { name: 'Mission 1', description: 'Explore the moon' },
            { name: 'Mission 2', description: 'Discover new galaxies' },
          ],
        },
        // Mock data for the second space cat
        {
          name: 'Steve',
          age: 5,
          missions: [
            { name: 'Mission 3', description: 'Study interstellar phenomena' },
            // You can add more missions as needed
          ],
        },
      ],
    }),

I got it running but you can see the definition is difference as i couldn't get the "missions" part to work/return in explorer when using the syntax of  spaceCats: () => [...new Array(6)], so captured the whole list of synthetic data in the [ ... ]. not sure why I couldn't get it working with the predefined property. Thanks.

**Comment by xuniaoji** [comment]
Hello, I got an issue with the typescript.
I am using the gql function which is generated by the codegen, 

Firstly I run the codegen to generate the graphql schema type

Then
```ts
// in request.ts file

import { gql } from '../@types/__generated__/gql';

export const getLoanCalculator = gql(/* GraphQL */ `
  query ExampleQuery($mockedStatus: String) {
      data: { XXXX }
  }
`

```
 there are two typescript-eslint errors:
'Unsafe assignment of an `any` value.eslint@typescript-eslint/no-unsafe-assignment'
`Unsafe call of an `any` typed value.eslint@typescript-eslint/no-unsafe-call`

Any suggestion for this?

**Reply by mabuyo** [reply]
Hey there @xuniaoji - it looks like this code is from your own personal project, and it's difficult for me to troubleshoot without seeing the full context. That said, do the documentation pages for those particular errors help you? 

- https://typescript-eslint.io/rules/no-unsafe-assignment/
- https://typescript-eslint.io/rules/no-unsafe-call/

**Comment by NishantCoder108** [comment]
# Defining a query , 🔥

## I learned
- for making query , first write query then queryName , and then curlybraces , in which you can define your query , what type of data you want exactly

```
query GetCatSpaceList {
      catSpace {
         name
         id
        image
 }
}

**Comment by virkumpavat** [comment]
No Errors So far

**Comment by Ankita-N715** [comment]
no errore
